### DISCUSSION TITLE
"ref" is getting null in all Quasar component

### DISCUSSION THREAD

**Discussion by bhavini-chauhan** [quasarframework/quasar#16930] [General - Components / Directives / etc] [answered]
Version of vue and Quasar:
"vue": "^3.0.0",
"quasar": "^2.5.5",

I am using "ref" in Quasar select component with composition API. 

```
<q-select v-model="model" :options="options" label="Standard" ref="selectRef" />

const selectRef = ref(null);

onMounted(() => {
  selectRef.value?.hidePopup();
})
```

While trying to get selectRef.value in onMounted or onBeforeMount. It is getting "null"
How to get "ref" value in Quasar component.

![image](https://github.com/quasarframework/quasar/assets/57613186/0da37b8c-caab-4595-ade0-08da3d8df86e)

**Comment by yusufkandemir** [comment, marked as answer]
It should normally work. But, it might be due to rendering order/conditions/etc., which could apply to any Vue component. So, you would need to share more of the `<template>` section for us to help you.

**Reply by bhavini-chauhan** [reply]
Thank you. It is solved. 
The problem was I haven't pass selectRef in return {}.
